enhancements/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,239 @@
+  - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+**Examples**: These are generalized examples to consider, in addition
+to the aforementioned [maturity levels][maturity-levels].
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+There is currently no active component managing the WorkloadPartitions custom resource. Since we do not want to allow enabling workload partitioning for existing clusters there does not have to be any upgrade logic.
+
+In the future, we will have to handle upgrade from older to newer versions of the WorkloadPartitions CRD if we change it in a backwards incompatible way.
+
+### Version Skew Strategy
+
+This CRD is closely tied to features in Kubelet and the new workload partitioning admission webhook. All those pieces are supposed to be shipped and deployed as part of standard OpenShift installation.
+
+In case an old kubelet with no admission webhook is used (hypothetical scenario) nothing will happen, the cluster will behave is if no workload partitioning was enabled
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+This feature and its implementation might lead cluster administrators to start using it for manual overrides of the core components like cpu and/or topology manager. That would be an anti-pattern and the presentation of this feature should clearly state that.
+
+## Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
+highlight and record other possible approaches to delivering the value proposed
+by an enhancement.
```

Comment:
Elsewhere we've discussed the possible need to add per-workload parameters, like scope. The proposal above implies those would end up as separate lists, which I think is fine. We should include the other form here in the Alternatives section, for completeness. Our notes doc has an example as

```yaml
workloadTypes:
    - name: user-defined-workload-type
      scope: Pool
    - name: management
      Scope: ClusterWide
```

Reply:
Good point!